lib/aggregate-translations.js

```diff
@@ -83,11 +85,19 @@ const aggregatedTranslations = (options) => {
   const outputDirectory = path.resolve(baseDir, outputDir);
   fileSystem.mkdirpSync(outputDirectory);
 
+  /**
+   * Detecting react-intl version used. `react-intl` v3 and greater does not export the intlShape.
+   * In order to use these versions of `react-intl` consumers should utilize terra-application v2, which provides the i18n loaders.
+   */
+  const isReactIntlv5 = !!intlShape;
+
   // Write aggregated translation messages to a file for each locale
-  writeAggregatedTranslations(aggregatedMessages, locales, fileSystem, outputDirectory);
+  writeAggregatedTranslations(aggregatedMessages, locales, fileSystem, outputDirectory, !isReactIntlv5);
 
-  // Write intl and translations loaders for the specified locales
-  writeI18nLoaders(locales, fileSystem, outputDirectory, format);
+  if (isReactIntlv5) {
```

Comment:
This seems to be working but I think I got myself turned around. Can you explain this a little more? In my head we should be doing the opposite. If we were writing the i18nloaders always in v2 and we upgrade to v5, I would think we'd still want to upgrade the loaders in v2 and maybe not in v5. So this logic seems backwards. We're writing the loaders in v5 but turning it off for existing v2 (when it was writing before in v2)?

Reply:
Maybe `isReactIntlv5` should be rename to `isReactIntlv2` to avoid the confusion because `!!intlShape` on line 92 would be `false` if `intlShape` is undefined...which means we are not using react-intl v2. If that why it is negated on line 95?

If we're on V5:
- const isReactIntlv5 = !!intlShape;   // !!undefined ===> false

If we're on V2:
- const isReactIntlv5 = !!intlShape;   // !!{} ===> true

Reply:
Yes, that is definitely misnamed.